### PR TITLE
[New Feature] Support triple grad in Paddle

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

 New features
### PR changes

OPs 
### Describe

This PR is trying to support higher order differentiation (3 order) in operators:

1. Support it by refactor rename strategy in double grad mechanism
2. Support triple grad gradient checker
3. Support Sigmoid triple grad kernel

### CODE REVIEW

File: python/paddle/fluid/backward.py [hunk 1]
```diff
@@ -408,7 +413,9 @@ def _strip_grad_suffix_(name):
     """
     name = cpt.to_text(name)
     pos = name.find(core.grad_var_suffix())
-    return name[:pos] if pos != -1 else name
+    new_name = name[:pos] if pos != -1 else name
+    new_pos = name.rfind('grad/')
+    return new_name[new_pos + 5:] if new_pos != -1 else new_name
```
Comment (chenwhql) [inline python/paddle/fluid/backward.py:418]: why add 5? do we need a comment here?

Comment (phlrain) [reply python/paddle/fluid/backward.py:418]: len('grad/') = 5

**Review by chenwhql — APPROVED** [review]
LGTM

File: python/paddle/fluid/backward.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/backward.py:418]: Sure, can I add comment in next pr

**Pull request merged**